src/pybind/mgr/volumes/fs/health.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import threading
+
+from .exception import VolumeException
+
+log = logging.getLogger(__name__)
+
+# singleton design pattern taken from http://www.aleax.it/5ep.html
+
+class HealthUpdateSingleton(object):
+    """
+    Singleton class to track health metadata for various "modules"
```

Comment:
Is this class meant for any 'mgr' module or for only 'volumes' mgr module?  If it's general purpose it doesn't make sense for it to require a 'volume_client' object, right?

Reply:
when it is moved to manager base, `volume_client` would not be required.